src/include/common/value_factory.h

```diff
@@ -63,15 +63,20 @@ class ValueFactory {
   }
 
   /// Constructs a value copied into long-lived pooled memory (or the heap)
-  /// that will require an explicit Value::free.
+  /// If the Varlen is constructed on the heap, then the object will
+  /// be freed automatically by the destructor inside Value's destructor
+  /// otherwise, it will be left in the string pool and does not require
+  /// deallocation
+  ///
+  /// Both cases do not need the caller to perform any special action such
+  /// as freeing memory
   static inline Value GetStringValue(const char *value,
                                      VarlenPool *pool = NULL) {
     return Value::GetAllocatedValue(VALUE_TYPE_VARCHAR, value,
                                     (size_t)(value ? strlen(value) : 0), pool);
```

Comment:
If you mean checking whether pool is nullptr -  it is done inside the Value object and is always checked